internal/common/function/function.go

```diff
@@ -77,21 +77,39 @@ func AddNonAdminBackupAnnotations(ownerNamespace string, ownerName string, owner
 	return mergedAnnotations
 }
 
+// containsOnlyNamespace checks if the given namespaces slice contains only the specified namespace
+func containsOnlyNamespace(namespaces []string, namespace string) bool {
+	for _, ns := range namespaces {
+		if ns != namespace {
+			return false
+		}
+	}
+	return true
+}
+
 // GetBackupSpecFromNonAdminBackup return BackupSpec object from NonAdminBackup spec, if no error occurs
 func GetBackupSpecFromNonAdminBackup(nonAdminBackup *nacv1alpha1.NonAdminBackup) (*velerov1api.BackupSpec, error) {
 	if nonAdminBackup == nil {
 		return nil, fmt.Errorf("nonAdminBackup is nil")
 	}
 
-	// TODO check spec?
-
 	if nonAdminBackup.Spec.BackupSpec == nil {
-		return nil, fmt.Errorf("BackupSpec is nil")
+		return nil, fmt.Errorf("BackupSpec is not defined")
 	}
 
+	veleroBackupSpec := nonAdminBackup.Spec.BackupSpec.DeepCopy()
+
 	// TODO: Additional validations, before continuing
 
-	return nonAdminBackup.Spec.BackupSpec.DeepCopy(), nil
+	if veleroBackupSpec.IncludedNamespaces == nil {
+		veleroBackupSpec.IncludedNamespaces = []string{nonAdminBackup.Namespace}
```

Comment:
does this change NonAdminBackup spec or just passes a different one to velero backup that will be created :question:

Reply:
So when there is no includenamespaces then it's changed otherwise it sets condition accepted to false.